validator/client/src/test/java/tech/pegasys/teku/validator/client/loader/PublicKeyLoaderTest.java

```diff
@@ -121,16 +123,18 @@ void shouldHandleEmptyResponseFromUrl() throws IOException {
   }
 
   @Test
-  void shouldThrowInvalidConfigurationExceptionWhenUrlFailsToLoad() throws Exception {
-    final UnknownHostException exception = new UnknownHostException("Unknown host");
-    when(mapper.readValue(new URL(urlSource), String[].class)).thenThrow(exception);
-    assertThatThrownBy(() -> loader.getPublicKeys(List.of(urlSource)))
-        .isInstanceOf(InvalidConfigurationException.class)
-        .hasRootCause(exception);
+  void shouldThrowExceptionWhenUrlFailsToLoad() {
+    final InvalidConfigurationException cause = new InvalidConfigurationException("Unknown host");
+    final CompletionException exception = new CompletionException(cause);
+    final ExternalUrlKeyReader reader = mock(ExternalUrlKeyReader.class);
+    when(reader.readKeys()).thenThrow(exception);
+
+    assertThatThrownBy(() -> loader.getPublicKeys(List.of(urlSource), (key) -> reader))
+        .isEqualTo(exception);
```

Comment:
I am confused by this test. It seems that all we want to test is that any exceptions bubble up to the caller of `PublicKeyLoader.getPublicKeys(..)`. If this is the case, we probably do not care about the exception type or cause. Might just use something simpler like:
```
final RuntimeException exception = new RuntimeException("Foo");
when(reader.readKeys()).thenThrow(exception);
```

Later we can test that the same exception is propagated up.

We could also rename the test to `shouldThrowExceptionWhenExternalUrlKeyReaderFailsUnexpectedly` to make it clear that this is a case where we do not handle the error.